### PR TITLE
Add `ConstOsmosisTestPool` and implement `lock_tokens` for `OsmosisTestRobot`

### PR DESCRIPTION
**Pull request by apollo-pacman**

### CODE REVIEW

File: src/osmosis/robot.rs [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
 };
 use osmosis_test_tube::{Account, OsmosisTestApp, Runner, SigningAccount};
 
 use crate::robot::TestRobot;
 
+/// Returns true if the provided denom is an follows the format of an Osmosis LP token
+fn is_osmosis_lp_token(denom: &str) -> bool {
+    let parts = denom.split('/').collect::<Vec<_>>();
+    parts[0] == "gamm" && parts[1] == "pool" && parts[2].parse::<u32>().is_ok()
+}
```
[the remark this answers is not in the document]
Comment (apollo-sturdy) [reply src/osmosis/robot.rs:20]: Might be worth adding a test too. E.g.
```rust
#[test]
fn test_is_osmosis_lp_token() {
    // Success cases
    assert!(is_osmosis_lp_token("gamm/pool/1"));
    assert!(is_osmosis_lp_token("gamm/pool/12"));
    assert!(is_osmosis_lp_token("gamm/pool/123"));
    assert!(is_osmosis_lp_token("gamm/pool/1234"));

    // Failure cases
    assert!(!is_osmosis_lp_token(""));
    assert!(!is_osmosis_lp_token("gamm"));
    assert!(!is_osmosis_lp_token("gamm/"));
    assert!(!is_osmosis_lp_token("gamm/pool"));
    assert!(!is_osmosis_lp_token("gamm/pool/"));
    assert!(!is_osmosis_lp_token("gamm/pool/1/"));
    assert!(!is_osmosis_lp_token("gamm/pool/1/2"));
    assert!(!is_osmosis_lp_token("gamm/pool/1/2/3"));
    assert!(!is_osmosis_lp_token("test/pool/1"));
    assert!(!is_osmosis_lp_token("gamm/test/1"));
    assert!(!is_osmosis_lp_token("gamm/pool/test"));
}
```

**Review by apollo-sturdy — CHANGES_REQUESTED** [review]

File: src/osmosis/robot.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (apollo-pacman) [reply src/osmosis/robot.rs:20]: fixed

File: src/osmosis/robot.rs [hunk 2]
```diff
@@ -111,6 +122,27 @@ pub trait OsmosisTestRobot<'a>: TestRobot<'a, OsmosisTestApp> {
 
         self
     }
+
+    /// Locks LP shares for a given duration in the osmosis lockup module
+    fn lock_tokens(&self, signer: &SigningAccount, coin: Coin, duration: i64) -> &Self {
```
[the remark this answers is not in the document]
Comment (apollo-pacman) [reply src/osmosis/robot.rs:127]: the message takes an i64 yes. but its a bit wierd. a u64 might not be castable to an i64 but i'll change it to an u32 instead which should be good

File: src/const_coin.rs [hunk 3]
```diff
@@ -1,5 +1,6 @@
 use cosmwasm_std::{Coin, Uint128};
 
+#[derive(Debug, Clone, PartialEq, Eq)]
```
[the remark this answers is not in the document]
Comment (apollo-pacman) [reply src/const_coin.rs:3]: done

**Pull request merged**